kclvm/tools/src/LSP/src/goto_def.rs

```diff
@@ -30,59 +37,212 @@ pub(crate) fn goto_definition(
     match program.pos_to_stmt(kcl_pos) {
         Some(node) => match node.node {
             Stmt::Import(stmt) => goto_def_for_import(&stmt, kcl_pos, prog_scope, program),
-            _ => {
-                let objs = find_definition_objs(node, kcl_pos, prog_scope);
-                let positions = objs
-                    .iter()
-                    .map(|obj| (obj.start.clone(), obj.end.clone()))
-                    .collect();
-                positions_to_goto_def_resp(&positions)
-            }
+            _ => match find_def(node.clone(), kcl_pos, prog_scope) {
+                Some(obj) => {
+                    let mut positions = IndexSet::new();
+                    positions.insert((obj.start.clone(), obj.end.clone()));
+                    return positions_to_goto_def_resp(&positions);
+                }
+                None => return None,
+            },
         },
         None => None,
     }
 }
-pub(crate) fn find_definition_objs(
+
+// Todo: fix ConfigExpr
+// ```kcl
+// schema Person:
+//     name: str
+//     data: Data
+
+// schema Data:
+//     id: int
+
+// person = Person {
+//     data.id = 1
+//     data: {
+//         id = 1
+//     }
+//     data: Data {
+//         id = 3
+//     }
+// }
+
+pub(crate) fn find_def(
     node: Node<Stmt>,
     kcl_pos: &KCLPos,
     prog_scope: &ProgramScope,
-) -> Vec<ScopeObject> {
+) -> Option<ScopeObject> {
     let (inner_expr, parent) = inner_most_expr_in_stmt(&node.node, kcl_pos, None);
     if let Some(expr) = inner_expr {
         if let Expr::Identifier(id) = expr.node {
```

Comment:
Use `impl TryInto<Node<Identifier>> for Node<Expr> {` defined in the `ast.rs`

Reply:
This should be match `expr.node`, but now only handled case of identifier